asserts/device_asserts.go

```diff
@@ -73,25 +73,40 @@ func (mod *Model) Architecture() string {
 	return mod.HeaderString("architecture")
 }
 
+// snapWithTrack represents a snap that includes optional track
```

Comment:
I didn't notice that we still have code about kernel-track to remove here

Reply:
I can refactor this if you prefer that this goes away now, e.g. I could make return Gadget/Kernel return a SnapWithTrack (we may need a better name for this). WDYT?